src/main/java/games/strategy/util/Version.java

```diff
@@ -204,7 +211,16 @@ public String toString() {
     return m_micro != 0 ? toStringFull() : m_major + "." + m_minor + (m_point != 0 ? "." + m_point : "");
   }
 
+  /**
+   * Indicates the specified version is compatible with this version.
+   *
+   * @param other The version to compare to this version for compatibility.
+   *
+   * @return {@code true} if the specified version is compatible with this version; otherwise {@code false}.
+   */
```

Comment:
We should perhaps mention, that passing null results in a NPE.
`throws NullPointerException When other is null`

Reply:
(a null value is obviously not compatible, but a programmer could expect it to return false similar to equals)

Reply:
There was a discussion on this topic a few months ago because I was going OCD on documenting nullity for method parameters and return values.  At that time, @DanVanAtta recommended that we adopt the convention that all parameters and return values are assumed to be non-`null` unless stated otherwise in the Javadocs, and that passing a `null` argument will result in an NPE.

Since that time, we've also introduced the JSR 305 nullity annotations.  Applying `@Nullable` to a method parameter or return type to indicate `null` is an acceptable value seems to be an improvement over explicitly writing it out in the Javadocs.  The lack of such an annotation on `other` should imply that passing `null` for that parameter will result in an NPE.